lib/google_crawler_web/controllers/upload_controller.ex

```diff
@@ -5,19 +5,55 @@ defmodule GoogleCrawlerWeb.UploadController do
   alias GoogleCrawler.Search
   alias GoogleCrawler.Search.KeywordFile
 
+  @spec create(atom | %{__struct__: atom}, map) :: Plug.Conn.t()
   def create(conn, %{"keyword_file" => keyword_file}) do
     changeset = KeywordFile.changeset(%KeywordFile{}, keyword_file)
 
     if changeset.valid? do
       file = get_change(changeset, :file, nil)
-      result = Search.parse_keywords_from_file!(file.path, file.content_type)
 
-      # TODO: Save these keywords and triggers the task to google search for each keyword
-      text(conn, result |> Enum.map(fn keyword -> List.first(keyword) end) |> Enum.join(", "))
+      Search.parse_keywords_from_file!(file.path, file.content_type)
+      |> create_and_trigger_google_search(conn)
+      |> put_error_flash_for_failed_keywords(conn)
+      |> redirect(to: Routes.dashboard_path(conn, :index))
     else
       conn
       |> put_flash(:error, gettext("Invalid file, please select again."))
       |> redirect(to: Routes.dashboard_path(conn, :index))
     end
   end
+
+  # TODO: Trigger the scrapper background worker
+  defp create_and_trigger_google_search(csv_result, conn) do
+    csv_result
+    |> Stream.map(fn keyword_row -> List.first(keyword_row) end)
+    |> Stream.map(fn keyword -> %{keyword: keyword} end)
+    |> Enum.map(&Search.create_keyword(&1, conn.assigns.current_user))
+  end
+
+  defp put_error_flash_for_failed_keywords(create_result, conn) do
```

Comment:
I'm displaying some error message when some of the keywords cannot be created but I'm not sure how to test it 🤔💭